src/main/scala/io/qbeast/spark/delta/OTreeIndex.scala

```diff
@@ -13,15 +13,20 @@ import org.apache.spark.sql.delta.actions.AddFile
 import org.apache.spark.sql.delta.files.TahoeLogFileIndex
 import org.apache.spark.sql.execution.datasources.{FileIndex, PartitionDirectory}
 import org.apache.spark.sql.types.StructType
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.execution.SQLExecution
 
 import java.net.URI
 
 /**
  * FileIndex to prune files
  *
  * @param index the Tahoe log file index
+ * @param spark spark session
  */
-case class OTreeIndex(index: TahoeLogFileIndex) extends FileIndex {
+case class OTreeIndex(index: TahoeLogFileIndex, spark: SparkSession)
```

Comment:
If we add `spark` to the parameters, we need to change it in any test or place where OTreeIndex is called. Maybe we can use `SparkSession.active ` to retrieve the context and use logging.

Reply:
Using SparkSession.active instead.